internal/controllers/topology/cluster/cluster_controller_test.go

```diff
@@ -226,6 +243,9 @@ func TestClusterReconciler_reconcileUpdateOnClusterTopology(t *testing.T) {
 		// Check if MachineDeployments are created and have the correct version, replicas, labels annotations and templates.
 		g.Expect(assertMachineDeploymentsReconcile(updatedCluster)).Should(Succeed())
 
+		// Check if MachinePools are created and have the correct version, replicas, labels annotations and templates.
+		g.Expect(assertMachinePoolsReconcile(actualCluster)).Should(Succeed())
```

Comment:
```suggestion
		g.Expect(assertMachinePoolsReconcile(updatedCluster)).Should(Succeed())
```

Reply:
Also we should add a equivalent to:

https://github.com/kubernetes-sigs/cluster-api/pull/9380/files#diff-76cc9048539f3091a1d62bd072d4d368008eb2016517e2aecc23c51eb26bf312R231-R232

```go
		// Check to ensure the replica count has been successfully updated in the API server and cache.
		g.Expect(updatedCluster.Spec.Topology.Workers.MachinePools[0].Replicas).To(Equal(&replicas))
```